src/puppetlabs/puppetdb/command.clj

```diff
@@ -131,27 +132,29 @@
           (string? mq-endpoint)
           (string? raw-command)]
    :post [(string? %)]}
+  ;; FIXME: is this efficient enough?  It's rebuilding the whole MQ
```

Comment:
@mullr Already fixed here -- I've reworked all of that side to share a single connection.